modules/tenant-management/src/Volo.Abp.TenantManagement.Domain/Volo/Abp/TenantManagement/TenantConfigurationCacheItemInvalidator.cs

```diff
@@ -22,12 +24,17 @@ public TenantConfigurationCacheItemInvalidator(IDistributedCache<TenantConfigura
 
     public virtual async Task HandleEventAsync(EntityChangedEventData<Tenant> eventData)
     {
+        if (eventData is EntityCreatedEventData<Tenant>)
+        {
+            return;
+        }
+
         await ClearCacheAsync(eventData.Entity.Id, eventData.Entity.NormalizedName);
     }
 
-    public virtual async Task HandleEventAsync(EntityDeletedEventData<Tenant> eventData)
+    public virtual async Task HandleEventAsync(TenantChangedEvent eventData)
```

Comment:
> ILocalEventHandler<EntityDeletedEventData<Tenant>>

I think the cache needs to be cleared when deleting a tenant.

Reply:
[EntityChangedEventData](https://github.com/abpframework/abp/blob/2ede81cb2d5326b01813e6ca54f5eb0ec5675310/framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/Events/EntityChangedEventData.cs#L6) enters for all changes Created, Updated, Deleted.